nntrainer/src/neuralnet.cpp

```diff
@@ -536,6 +529,9 @@ sharedTensor NeuralNetwork::forwarding(sharedTensor input) {
 sharedTensor NeuralNetwork::forwarding(sharedTensor input, sharedTensor label) {
   sharedTensor X;
 
+  if (input->getDim().batch() > batch_size)
+    throw std::logic_error("Error: mismatch in batchsize for data and model.");
```

Comment:
Could you add `std::logic_error` to `nntrainer_exception_boundary`?

Reply:
done